gnovm/pkg/gnolang/preprocess.go

```diff
@@ -81,15 +81,40 @@ func PredefineFileSet(store Store, pn *PackageNode, fset *FileSet) {
 	// Finally, predefine other decls and
 	// preprocess ValueDecls..
 	for _, fn := range fset.Files {
-		for i := 0; i < len(fn.Decls); i++ {
+		lenDecls := len(fn.Decls)
+		for i := 0; i < lenDecls; i++ {
 			d := fn.Decls[i]
 			if d.GetAttribute(ATTR_PREDEFINED) == true {
 				// skip declarations already predefined (e.g.
 				// through recursion for a dependent)
 			} else {
-				// recursively predefine dependencies.
-				d2, _ := predefineNow(store, fn, d)
-				fn.Decls[i] = d2
+				if vd, ok := d.(*ValueDecl); ok && len(vd.NameExprs) > 1 && len(vd.Values) == len(vd.NameExprs) {
```

Comment:
we should also need handle this:
package main

func main() {
	println(a)
	println(b)
	println(c)
}

var a, b, c = 1, a + 1

len(Lhs) != len(Rhs), but it can be another issue.

Reply:
Yes, we should handle this case. However, it is not related to the problem this PR is solving. We should do it in another issue + PR

Reply:
@ltzmaxwell can you open an issue on this?

cc @Kouteki

Reply:
https://github.com/gnolang/gno/issues/2137 @zivkovicmilos @Kouteki @petar-dambovaliev